diff -Naur clang-2.9//lib/Frontend/InitHeaderSearch.cpp clang-2.9-patched//lib/Frontend/InitHeaderSearch.cpp
--- clang-2.9//lib/Frontend/InitHeaderSearch.cpp	2011-03-22 00:24:04.000000000 +0000
+++ clang-2.9-patched//lib/Frontend/InitHeaderSearch.cpp	2011-05-06 01:01:11.000000000 +0000
@@ -748,10 +748,10 @@
     AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.5",
                                 "x86_64-suse-linux", "", "", triple);
 
-    // Arch Linux 2008-06-24
-    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.3.1",
+    // Chakra Linux 2011-05-06
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.5.3",
                                 "i686-pc-linux-gnu", "", "", triple);
-    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.3.1",
+    AddGnuCPlusPlusIncludePaths("/usr/include/c++/4.5.3",
                                 "x86_64-unknown-linux-gnu", "", "", triple);
     // Gentoo x86 gcc 4.5.2
     AddGnuCPlusPlusIncludePaths(
