diff --git a/localedata/locales/bo_CN b/localedata/locales/bo_CN
index aa8ff07..e568280 100644
--- a/localedata/locales/bo_CN
+++ b/localedata/locales/bo_CN
@@ -145,7 +145,7 @@  END LC_MEASUREMENT
 LC_NAME
 % FIXME

-name_fmt  ""
+name_fmt  "???"
 % name_gen	"FIXME"
 % name_miss	"FIXME"
 % name_mr	"FIXME"
diff --git a/localedata/locales/bo_IN b/localedata/locales/bo_IN
index 9e9c4ff..16e8dd3 100644
--- a/localedata/locales/bo_IN
+++ b/localedata/locales/bo_IN
@@ -70,7 +70,7 @@  END LC_MEASUREMENT

 LC_NAME
 % FIXME
-name_fmt	""
+name_fmt	"???"
 % name_gen	"FIXME"
 % name_miss	"FIXME"
 % name_mr	"FIXME"

--
