--- src/gwenhywfar-4.13.1/src/sio/syncio_tls.c  2015-09-13 11:53:57.116666673 +0000
+++ src/gwenhywfar-4.13.1/src/sio/syncio_tls.c  2015-09-13 11:53:57.116666673 +0000
@@ -366,7 +366,7 @@
 
     DBG_INFO(GWEN_LOGDOMAIN, "Forcing SSL v3");
-    rv=gnutls_protocol_set_priority(xio->session, proto_prio);
+    rv=gnutls_priority_set_direct(xio->session, ":-VERS-TLS-ALL:+VERS-SSL3.0", NULL);
     if (rv) {
       DBG_ERROR(GWEN_LOGDOMAIN, "gnutls_protocol_set_priority: %d (%s)", rv, gnutls_strerror(rv));
       gnutls_deinit(xio->session);

