---
 src/gui/text/qtextengine.cpp |    1 +
 1 file changed, 1 insertion(+)

--- a/src/gui/text/qtextengine.cpp
+++ b/src/gui/text/qtextengine.cpp
@@ -1640,6 +1640,7 @@ void QTextEngine::itemize() const
             case QChar::Script_Katakana:
             case QChar::Script_Bopomofo:
             case QChar::Script_Han:
+            case QChar::Script_Gujarati:
                 analysis[i].script = QChar::Script_Common;
                 break;
             default:
