diff -aur bla/src/wl/sys/wl_linux.c bla.new/src/wl/sys/wl_linux.c
--- bla/src/wl/sys/wl_linux.c	2010-02-06 02:59:15.000000000 +0100
+++ bla.new/src/wl/sys/wl_linux.c	2010-08-07 23:17:35.368641722 +0200
@@ -187,6 +187,8 @@
 
 static int nompc = 0;
 module_param(nompc, int, 0);
+MODULE_LICENSE("Mixed/Proprietary"); 
+
 
 static char name[IFNAMSIZ] = "eth%d";
 module_param_string(name, name, IFNAMSIZ, 0);
