diff --git a/Source/WebKit/qt/WebCoreSupport/NotificationPresenterClientQt.cpp b/Source/WebKit/qt/WebCoreSupport/NotificationPresenterClientQt.cpp
index 56ed0b4..1e50c9e 100644
--- a/Source/WebKit/qt/WebCoreSupport/NotificationPresenterClientQt.cpp
+++ b/Source/WebKit/qt/WebCoreSupport/NotificationPresenterClientQt.cpp
@@ -323,7 +323,8 @@ void NotificationPresenterClientQt::requestPermission(ScriptExecutionContext* co
         RefPtr<VoidCallback> cb = callback;
         CallbacksInfo info;
         info.m_frame = toFrame(context);
-        info.m_callbacks.append(cb);
+        if (cb)
+            info.m_callbacks.append(cb);
         m_pendingPermissionRequests.insert(context, info);
 
         if (toPage(context) && toFrame(context)) {
         