diff -up libkscreen-1.0.5/CMakeLists.txt.VERSION_RELEASE libkscreen-1.0.5/CMakeLists.txt
--- libkscreen-1.0.5/CMakeLists.txt.VERSION_RELEASE	2014-08-03 16:14:49.000000000 -0500
+++ libkscreen-1.0.5/CMakeLists.txt	2014-10-31 09:32:07.361195742 -0500
@@ -1,7 +1,7 @@
 project(libkscreen)
 set(libkscreen_VERSION_MAJOR 1)
 set(libkscreen_VERSION_MINOR 0)
-set(libkscreen_VERSION_RELEASE 4)
+set(libkscreen_VERSION_RELEASE 5)
 set(libkscreen_VERSION "${libkscreen_VERSION_MAJOR}.${libkscreen_VERSION_MINOR}.${libkscreen_VERSION_RELEASE}")
 add_definitions("-DLIBKSCREEN_VERSION=\\\"${libkscreen_VERSION}\\\"")
 
