--- e4rat-0.2.3/src/logging.hh.orig	2017-03-17 15:15:58.603287712 +0000
+++ e4rat-0.2.3/src/logging.hh	2017-03-17 15:16:12.265147263 +0000
@@ -93,7 +93,7 @@
 #define dump_log(...) logger.write(__VA_ARGS__)
 
 #ifdef DEBUG_ENABLED
-    #define debug(format,args...) dump_log(Debug, "%s:%d in %s(): "format, __FILE__, __LINE__, __FUNCTION__, ## args)
+    #define debug(format,args...) dump_log(Debug, "%s:%d in %s(): " format, __FILE__, __LINE__, __FUNCTION__, ## args)
 #else
     #define debug(format,args...)
 #endif
