diff --git a/src/service/ksmserver/KSMServer.cpp b/src/service/ksmserver/KSMServer.cpp
index b5e1467..695f642 100644
--- a/src/service/ksmserver/KSMServer.cpp
+++ b/src/service/ksmserver/KSMServer.cpp
@@ -69,7 +69,7 @@ void KSMServer::Private::serviceOwnerChanged(const QString &service,
 
         if (KDBusConnectionPool::threadConnection().interface()->isServiceRegistered(KWIN_SERVICE)) {
             // Creating the new dbus interface
-            kwin = new QDBusInterface(KWIN_SERVICE, QStringLiteral("/KWin"), QStringLiteral("org.kde.kwin"));
+            kwin = new QDBusInterface(KWIN_SERVICE, QStringLiteral("/KWin"), QStringLiteral("org.kde.KWin"));
 
             // If the service is valid, initialize it
             // otherwise delete the object
@@ -178,6 +178,8 @@ void KSMServer::Private::startCallFinished(QDBusPendingCallWatcher *call)
 
         if (!retval) {
             subSessionSendEvent(KSMServer::Stopped);
+        } else {
+            subSessionSendEvent(KSMServer::Started);
         }
     }
 
@@ -198,6 +200,8 @@ void KSMServer::Private::stopCallFinished(QDBusPendingCallWatcher *call)
 
         if (!retval) {
             subSessionSendEvent(KSMServer::FailedToStop);
+        } else {
+            subSessionSendEvent(KSMServer::Stopped);
         }
     }
 
@@ -214,18 +218,3 @@ void KSMServer::Private::subSessionSendEvent(int event)
 
     processingActivity.clear();
 }
-
-void KSMServer::Private::subSessionOpened()
-{
-    subSessionSendEvent(KSMServer::Started);
-}
-
-void KSMServer::Private::subSessionClosed()
-{
-    subSessionSendEvent(KSMServer::Stopped);
-}
-
-void KSMServer::Private::subSessionCloseCanceled()
-{
-    subSessionSendEvent(KSMServer::FailedToStop);
-}
diff --git a/src/service/ksmserver/KSMServer_p.h b/src/service/ksmserver/KSMServer_p.h
index c0f5598..86153d8 100644
--- a/src/service/ksmserver/KSMServer_p.h
+++ b/src/service/ksmserver/KSMServer_p.h
@@ -50,9 +50,6 @@ private Q_SLOTS:
     void startCallFinished(QDBusPendingCallWatcher *watcher);
     void stopCallFinished(QDBusPendingCallWatcher *watcher);
 
-    void subSessionOpened();
-    void subSessionClosed();
-    void subSessionCloseCanceled();
     void subSessionSendEvent(int event);
 
 private:
