diff -Naur openjdk.orig/corba/make/common/shared/Defs-utils.gmk openjdk/corba/make/common/shared/Defs-utils.gmk
--- openjdk.orig/corba/make/common/shared/Defs-utils.gmk	2012-08-11 01:26:31.654997270 +0000
+++ openjdk/corba/make/common/shared/Defs-utils.gmk	2012-08-11 01:30:52.550250600 +0000
@@ -131,7 +131,7 @@
 TAIL           = $(UTILS_USR_BIN_PATH)tail
 TAR            = $(UTILS_COMMAND_PATH)tar
 TEST           = $(UTILS_USR_BIN_PATH)test
-TOUCH          = $(UTILS_USR_BIN_PATH)touch
+TOUCH          = touch
 TR             = $(UTILS_USR_BIN_PATH)tr
 TRUE           = $(UTILS_COMMAND_PATH)true
 UNAME          = $(UTILS_COMMAND_PATH)uname
@@ -218,7 +218,7 @@
   SED          = $(UTILS_USR_BIN_PATH)sed
   SORT         = $(UTILS_USR_BIN_PATH)sort
   TEST         = $(UTILS_COMMAND_PATH)test
-  TOUCH        = $(UTILS_USR_BIN_PATH)touch
+  TOUCH        = touch
   TRUE         = $(UTILS_USR_BIN_PATH)true
   UNAME        = $(UTILS_USR_BIN_PATH)uname
   # BSD OS_VENDOR specific
diff -Naur openjdk.orig/jdk/make/common/shared/Defs-utils.gmk openjdk/jdk/make/common/shared/Defs-utils.gmk
--- openjdk.orig/jdk/make/common/shared/Defs-utils.gmk	2012-08-11 01:26:31.191666448 +0000
+++ openjdk/jdk/make/common/shared/Defs-utils.gmk	2012-08-11 01:30:52.546917344 +0000
@@ -148,7 +148,7 @@
 TAIL           = $(UTILS_USR_BIN_PATH)tail
 TAR            = $(UTILS_COMMAND_PATH)tar
 TEST           = $(UTILS_USR_BIN_PATH)test
-TOUCH          = $(UTILS_USR_BIN_PATH)touch
+TOUCH          = touch
 TR             = $(UTILS_USR_BIN_PATH)tr
 TRUE           = $(UTILS_COMMAND_PATH)true
 UNAME          = $(UTILS_COMMAND_PATH)uname
@@ -203,7 +203,7 @@
   # others have it in /usr/bin.
   SORT=$(firstword $(wildcard $(UTILS_COMMAND_PATH)sort) \
 		   $(wildcard $(UTILS_USR_BIN_PATH)sort))
-  NAWK           = $(USRBIN_PATH)gawk
+  NAWK           = gawk
   # Intrinsic unix command, with backslash-escaped character interpretation
   ECHO           = /bin/echo -e
   # These are really in UTILS_USR_BIN_PATH on Linux
@@ -240,7 +240,7 @@
   SED          = $(UTILS_USR_BIN_PATH)sed
   SORT         = $(UTILS_USR_BIN_PATH)sort
   TEST         = $(UTILS_COMMAND_PATH)test
-  TOUCH        = $(UTILS_USR_BIN_PATH)touch
+  TOUCH        = touch
   TRUE         = $(UTILS_USR_BIN_PATH)true
   UNAME        = $(UTILS_USR_BIN_PATH)uname
   NAWK         = $(UTILS_USR_BIN_PATH)awk
