=== modified file 'util/grub.d/10_kfreebsd.in'
--- util/grub.d/10_kfreebsd.in	2012-09-18 11:04:06 +0000
+++ util/grub.d/10_kfreebsd.in	2012-11-19 19:55:17 +0000
@@ -98,7 +98,7 @@
   fi
 
   printf '%s\n' "${prepare_boot_cache}" | sed "s/^/$submenu_indentation/"
-  message="$(gettext_printf "Loading kernel of FreeBSD %s ..." ${version})"
+  message="$(gettext_printf "Loading kernel of FreeBSD %s ..." ${version} | grub_quote)"
   sed "s/^/$submenu_indentation/" << EOF
 	echo			'$(echo "$message" | grub_quote)'
 	kfreebsd		${rel_dirname}/${basename} ${args}

=== modified file 'util/grub.d/20_linux_xen.in'
--- util/grub.d/20_linux_xen.in	2012-09-18 11:04:06 +0000
+++ util/grub.d/20_linux_xen.in	2012-11-19 19:55:17 +0000
@@ -117,8 +117,8 @@
     prepare_boot_cache="$(prepare_grub_to_access_device ${GRUB_DEVICE_BOOT} | grub_add_tab)"
   fi
   printf '%s\n' "${prepare_boot_cache}" | sed "s/^/$submenu_indentation/"
-  xmessage="$(gettext_printf "Loading Xen %s ..." ${xen_version})"
-  lmessage="$(gettext_printf "Loading Linux %s ..." ${version})"
+  xmessage="$(gettext_printf "Loading Xen %s ..." ${xen_version} | grub_quote)"
+  lmessage="$(gettext_printf "Loading Linux %s ..." ${version} | grub_quote)"
   sed "s/^/$submenu_indentation/" << EOF
 	echo	'$(echo "$xmessage" | grub_quote)'
 	multiboot	${rel_xen_dirname}/${xen_basename} placeholder ${xen_args}
@@ -127,7 +127,7 @@
 EOF
   if test -n "${initrd}" ; then
     # TRANSLATORS: ramdisk isn't identifier. Should be translated.
-    message="$(gettext_printf "Loading initial ramdisk ...")"
+    message="$(gettext_printf "Loading initial ramdisk ..." | grub_quote)"
     sed "s/^/$submenu_indentation/" << EOF
 	echo	'$(echo "$message" | grub_quote)'
 	module	${rel_dirname}/${initrd}

