--- mlt-0.9.6/src/modules/frei0r/factory.c.orig	2015-04-16 05:51:24.080000003 +0000
+++ mlt-0.9.6/src/modules/frei0r/factory.c	2015-04-16 05:52:20.310000005 +0000
@@ -40,7 +40,7 @@
 #define FREI0R_PLUGIN_PATH "/lib/frei0r-1"
 #else
 #define LIBSUF ".so"
-#define FREI0R_PLUGIN_PATH "/usr/lib/frei0r-1:/usr/lib64/frei0r-1:/opt/local/lib/frei0r-1:/usr/local/lib/frei0r-1:$HOME/.frei0r-1/lib"
+#define FREI0R_PLUGIN_PATH "/opt/opencv-qt5/lib/frei0r-1:/usr/local/lib/frei0r-1:$HOME/.frei0r-1/lib"
 #endif
 
 #define GET_FREI0R_PATH (getenv("FREI0R_PATH") ? getenv("FREI0R_PATH") : getenv("MLT_FREI0R_PLUGIN_PATH") ? getenv("MLT_FREI0R_PLUGIN_PATH") : FREI0R_PLUGIN_PATH)
