--- pulsesupport.cpp.orig	2013-12-06 18:22:44.000000000 +0800
+++ pulsesupport.cpp	2013-12-11 17:50:17.013087219 +0800
@@ -1332,7 +1332,7 @@
             return false;
         }
         pa_operation_unref(o);
-    } else if (s_outputStreams.contains(streamUuid) && s_outputStreams[streamUuid]->index() == PA_INVALID_INDEX) {
+    } else if (s_outputStreams.contains(streamUuid) && s_outputStreams[streamUuid]->index() == PA_INVALID_INDEX && !Phonon::NotificationCategory) {
         logMessage(QString::fromLatin1("Setting volume on an invalid stream ..... this better be intended"));
         PulseStream *stream = s_outputStreams[streamUuid];
         stream->setCachedVolume(volume);
