--- frei0r-plugins-1.4/src/Makefile.am	2015-04-16 05:38:44.453333338 +0000
+++ frei0r-plugins-1.4/src/Makefile.am.orig	2015-04-16 05:39:04.930000005 +0000
@@ -149,11 +149,13 @@
 if HAVE_OPENCV
 plugin_LTLIBRARIES += facebl0r.la
 facebl0r_la_SOURCES = filter/facebl0r/facebl0r.cpp
-facebl0r_la_CFLAGS = @OPENCV_CFLAGS@ @CFLAGS@ 
+facebl0r_la_CXXFLAGS = @OPENCV_CFLAGS@ @CXXFLAGS@ 
+facebl0r_la_LDFLAGS = -R /opt/opencv-qt5/lib -module -avoid-version -lm -export-dynamic
 facebl0r_la_LIBADD = @OPENCV_LIBS@
 plugin_LTLIBRARIES += facedetect.la
 facedetect_la_SOURCES = filter/facedetect/facedetect.cpp
-facedetect_la_CFLAGS = @OPENCV_CFLAGS@ @CFLAGS@
+facedetect_la_CXXFLAGS = @OPENCV_CFLAGS@ @CXXFLAGS@
+facedetect_la_LDFLAGS = -R /opt/opencv-qt5/lib -module -avoid-version -lm -export-dynamic
 facedetect_la_LIBADD = @OPENCV_LIBS@
 endif
 
--- frei0r-plugins-1.4/configure.ac.orig	2015-04-16 06:18:39.586666672 +0000
+++ frei0r-plugins-1.4/configure.ac	2015-04-16 06:19:48.533333338 +0000
@@ -5,7 +5,7 @@
 AC_INIT(frei0r-plugins, [1.3.0], [frei0r-devel@piksel.no])
 AC_CONFIG_MACRO_DIR([m4])
 
-AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
+AM_INIT_AUTOMAKE([subdir-objects])
 
 # Checks for programs.
 AC_PROG_CXX
