diff --git a/CMakeLists.txt b/CMakeLists.txt
index 09315ad..1dc4a80 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,6 @@ find_package(Plymouth REQUIRED)
 set(CMAKE_SHARED_MODULE_PREFIX "")
 
 add_subdirectory(breeze-text)
-install(DIRECTORY breeze DESTINATION ${Plymouth_THEMESDIR})
+add_subdirectory(breeze)
 
 feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)
diff --git a/breeze/CMakeLists.txt b/breeze/CMakeLists.txt
new file mode 100644
index 0000000..559c7fb
--- /dev/null
+++ b/breeze/CMakeLists.txt
@@ -0,0 +1,8 @@
+configure_file(breeze.plymouth.cmake ${CMAKE_CURRENT_BINARY_DIR}/breeze.plymouth)
+
+install(FILES breeze.grub
+              breeze.script
+              ${CMAKE_CURRENT_BINARY_DIR}/breeze.plymouth
+        DESTINATION ${Plymouth_THEMESDIR}/breeze)
+install(DIRECTORY images
+        DESTINATION ${Plymouth_THEMESDIR}/breeze)
diff --git a/breeze/breeze.plymouth b/breeze/breeze.plymouth
deleted file mode 100644
index d31431e..0000000
--- a/breeze/breeze.plymouth
+++ /dev/null
@@ -1,8 +0,0 @@
-[Plymouth Theme]
-Name=Breeze
-Description=A Breeze Splash
-ModuleName=script
-
-[script]
-ImageDir=/lib/plymouth/themes/breeze
-ScriptFile=/lib/plymouth/themes/breeze/breeze.script
diff --git a/breeze/breeze.plymouth.cmake b/breeze/breeze.plymouth.cmake
new file mode 100644
index 0000000..29e13ed
--- /dev/null
+++ b/breeze/breeze.plymouth.cmake
@@ -0,0 +1,9 @@
+[Plymouth Theme]
+Name=Breeze
+Description=A Breeze Splash
+ModuleName=script
+
+[script]
+ImageDir=@Plymouth_THEMESDIR@/breeze
+ScriptFile=@Plymouth_THEMESDIR@/breeze/breeze.script
+
