diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1dc4a80..1a6acad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,6 +11,8 @@ include(FeatureSummary)
 
 find_package(Plymouth REQUIRED)
 
+find_library(MATH_LIBRARY m)
+
 # Otherwise MODULE libs have a lib prefix, which is not compatible with
 # plymouth plugin names (i.e we want breeze-text.so not libbreeze-text.so).
 set(CMAKE_SHARED_MODULE_PREFIX "")
diff --git a/breeze-text/CMakeLists.txt b/breeze-text/CMakeLists.txt
index 17501cb..ce8d62a 100644
--- a/breeze-text/CMakeLists.txt
+++ b/breeze-text/CMakeLists.txt
@@ -1,5 +1,5 @@
 add_library(breeze-text MODULE plugin.c ply-text-progress-bar.c)
-target_link_libraries(breeze-text ${Plymouth_LIBRARIES})
+target_link_libraries(breeze-text ${Plymouth_LIBRARIES} ${MATH_LIBRARY})
 target_include_directories(breeze-text PRIVATE ${Plymouth_INCLUDE_DIRS})
 install(TARGETS breeze-text DESTINATION ${Plymouth_PLUGINSDIR})
 
