From 9711a01d380627d1e1535f1bcd0d74fff1476a45 Mon Sep 17 00:00:00 2001
From: Phillip Lougher <phillip@lougher.demon.co.uk>
Date: Sun, 06 Dec 2009 02:15:08 +0000
Subject: Squashfs: select DECOMPRESS_LZMA_NEEDED when including support for lzma

Signed-off-by: Phillip Lougher <phillip@lougher.demon.co.uk>
---
diff --git a/fs/squashfs/Kconfig b/fs/squashfs/Kconfig
index 5632f2a..6fca7b7 100644
--- a/fs/squashfs/Kconfig
+++ b/fs/squashfs/Kconfig
@@ -41,6 +41,7 @@ config SQUASHFS_LZMA
 	bool "Include support for LZMA compressed file systems"
 	depends on SQUASHFS
 	select DECOMPRESS_LZMA
+	select DECOMPRESS_LZMA_NEEDED
 	  
 
 config SQUASHFS_LZO
--
cgit v0.8.3.1
