From b5ff8407b58cf4128c1a7edd63c4155aa7724754 Mon Sep 17 00:00:00 2001
From: Phillip Lougher <phillip@lougher.demon.co.uk>
Date: Thu, 06 May 2010 01:00:11 +0000
Subject: squashfs: add missing slab.h include in lzma_wrapper

Signed-off-by: Phillip Lougher <phillip@lougher.demon.co.uk>
---
diff --git a/fs/squashfs/lzma_wrapper.c b/fs/squashfs/lzma_wrapper.c
index cef06d6..4438475 100644
--- a/fs/squashfs/lzma_wrapper.c
+++ b/fs/squashfs/lzma_wrapper.c
@@ -22,6 +22,7 @@
  */
 
 #include <asm/unaligned.h>
+#include <linux/slab.h>
 #include <linux/buffer_head.h>
 #include <linux/mutex.h>
 #include <linux/vmalloc.h>
--
cgit v0.8.3.1
