--- linux-3.18/fs/aufs/debug.c.orig	2014-12-18 20:26:05.364634951 +0000
+++ linux-3.18/fs/aufs/debug.c	2014-12-18 20:26:57.145531475 +0000
@@ -169,7 +169,7 @@
 	struct dentry *d;
 
 	spin_lock(&inode->i_lock);
-	hlist_for_each_entry(d, &inode->i_dentry, d_alias)
+	hlist_for_each_entry(d, &inode->i_dentry, d_u.d_alias)
 		au_dpri_dentry(d);
 	spin_unlock(&inode->i_lock);
 }
--- linux-3.18/fs/aufs/hnotify.c.orig	2014-12-18 20:25:18.717161233 +0000
+++ linux-3.18/fs/aufs/hnotify.c	2014-12-18 20:26:23.324945835 +0000
@@ -211,7 +211,7 @@
 		AuDebugOn(!name);
 		au_iigen_dec(inode);
 		spin_lock(&inode->i_lock);
-		hlist_for_each_entry(d, &inode->i_dentry, d_alias) {
+		hlist_for_each_entry(d, &inode->i_dentry, d_u.d_alias) {
 			spin_lock(&d->d_lock);
 			dname = &d->d_name;
 			if (dname->len != nlen
@@ -378,7 +378,7 @@
 
 	dentry = NULL;
 	spin_lock(&parent->d_lock);
-	list_for_each_entry(d, &parent->d_subdirs, d_u.d_child) {
+	list_for_each_entry(d, &parent->d_subdirs, d_child) {
 		/* AuDbg("%pd\n", d); */
 		spin_lock_nested(&d->d_lock, DENTRY_D_LOCK_NESTED);
 		dname = &d->d_name;
--- linux-3.18/fs/aufs/dcsub.c.orig	2014-12-18 20:25:22.400558273 +0000
+++ linux-3.18/fs/aufs/dcsub.c	2014-12-18 20:26:43.728632444 +0000
@@ -134,7 +134,7 @@
 	while (next != &this_parent->d_subdirs) {
 		struct list_head *tmp = next;
 		struct dentry *dentry = list_entry(tmp, struct dentry,
-						   d_u.d_child);
+						   d_child);
 
 		next = tmp->next;
 		spin_lock_nested(&dentry->d_lock, DENTRY_D_LOCK_NESTED);
@@ -170,7 +170,7 @@
 		this_parent = tmp;
 		spin_lock(&this_parent->d_lock);
 		rcu_read_unlock();
-		next = child->d_u.d_child.next;
+		next = child->d_child.next;
 		goto resume;
 	}
 
--- linux-3.18/fs/aufs/export.c.orig	2014-12-18 20:26:02.147912614 +0000
+++ linux-3.18/fs/aufs/export.c	2014-12-18 21:38:45.436937505 +0000
@@ -243,7 +243,7 @@
 		dentry = d_find_alias(inode);
 	else {
 		spin_lock(&inode->i_lock);
-		hlist_for_each_entry(d, &inode->i_dentry, d_alias) {
+		hlist_for_each_entry(d, &inode->i_dentry, d_u.d_alias) {
 			spin_lock(&d->d_lock);
 			if (!au_test_anon(d)
 			    && d->d_parent->d_inode->i_ino == dir_ino) {
