diff -Naur cmake-2.8.10.2-orig/Modules/FindGTK2.cmake cmake-2.8.10.2/Modules/FindGTK2.cmake
--- cmake-2.8.10.2-orig/Modules/FindGTK2.cmake	2012-11-27 14:26:32.000000000 +0100
+++ cmake-2.8.10.2/Modules/FindGTK2.cmake	2013-10-06 10:38:08.368594779 +0200
@@ -218,6 +218,8 @@
             /usr/pkg/include/glib
             $ENV{GTKMM_BASEPATH}/include
             $ENV{GTKMM_BASEPATH}/lib
+            /extra/usr/include
+            /extra/usr/lib
             [HKEY_CURRENT_USER\\SOFTWARE\\gtkmm\\2.4;Path]/include
             [HKEY_CURRENT_USER\\SOFTWARE\\gtkmm\\2.4;Path]/lib
             [HKEY_LOCAL_MACHINE\\SOFTWARE\\gtkmm\\2.4;Path]/include
